gax/src/main/java/com/google/api/gax/retrying/RetryingContext.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.retrying;
+
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalExtensionOnly;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+
+/**
+ * Context for a retryable operation.
+ *
+ * <p>It provides state to individual {@link RetryingFuture}s via the {@link RetryingExecutor}.
+ */
+@BetaApi
+@InternalExtensionOnly
```

Comment:
From `@InternalExtensionOnly`'s doc:  

"Indicates a public API that is stable for callers to use"

while `BetaApi`: 

"Indicates a public API that can change at any time, and has no guarantee of API stability" 

which makes them mutually exclusive (I understand that having both makes it harder to forget to add `@InternalExtensionOnly` when removing `@BetaApi`, but we need to either change docs in those annotations or remove `@InternalExtensionOnly` here (and other similar places)).

Reply:
It can make sense to use them together when you have functionality that isn't stable yet, but when you *do* stabilize it, you still don't want consumers extending it.

Reply:
I can change internal extension only into a comment

Reply:
Actually, looking at the docs for BetaApi, it seems like it expects itself to be mixed with InternalExtensionOnly:

https://github.com/googleapis/api-common-java/blob/master/src/main/java/com/google/api/core/BetaApi.java#L50-L51.

So I think it makes sense to leave it

Reply:
Ok